app/src/main/java/com/example/shorebuddy/data/lakes/Lake.java

```diff
@@ -1,13 +1,60 @@
 package com.example.shorebuddy.data.lakes;
 
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+import androidx.room.TypeConverters;
+
+import com.example.shorebuddy.utilities.Converters;
 import com.google.android.gms.maps.model.LatLng;
 
+import org.jetbrains.annotations.NotNull;
+
 // TODO: Implement as a Room model
+@Entity(tableName = "lake_table")
 public class Lake {
+
+    @PrimaryKey
+    @NotNull
     public final String name;
-    public LatLng location = new LatLng(0, 0);
 
-    public Lake(String n) {
-        name = n;
+    @ColumnInfo(name = "Coordinates")
+    @TypeConverters(Converters.class)
+    public LatLng location = new LatLng(0,0);
+
+
+    public Lake(@NotNull String name) {
+        this.name = name;
     }
+
+    @NotNull
+    public String getLakeName(){ return this.name;}
+
+    @NotNull
+    public LatLng getLakeCoor(){return this.location;}
+
+
+//    public Lake(String name, int elevation, int bassNumber){
+//
+//    }
+//    @ColumnInfo(name = "Coordinates")
+//    public LatLng location = new LatLng(0, 0);
+    @ColumnInfo(name = "Elevation")
+    public double elevation;
+
+    @ColumnInfo(name = "Bass")
+    public int numberBass;
```

Comment:
These field names could be simplified to bass, catfish, adSalmon, etc. Looking at the lakes csv file, it seems like these columns represent whether or not the species is found in the lake (0 not found, 1 found), not the amount of fish in the lake.

Reply:
Thank you for the clarifications and the recommendations, I made the changes for this section of the code